spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/rxjava/SleuthRxJavaSchedulersHook.java

```diff
@@ -77,16 +84,29 @@ public Action0 onSchedule(Action0 action) {
 		private Tracer tracer;
 		private TraceKeys traceKeys;
 		private Span parent;
+		private final List<String> threadsToIgnore;
 
-		public TraceAction(Tracer tracer, TraceKeys traceKeys, Action0 actual) {
+		public TraceAction(Tracer tracer, TraceKeys traceKeys, Action0 actual,
+				List<String> threadsToIgnore) {
 			this.tracer = tracer;
 			this.traceKeys = traceKeys;
+			this.threadsToIgnore = threadsToIgnore;
 			this.parent = tracer.getCurrentSpan();
 			this.actual = actual;
 		}
 
+		@SuppressWarnings("Duplicates")
 		@Override
 		public void call() {
+			// don't create a span if the thread name is on a list of threads to ignore
+			for (String threadToIgnore : this.threadsToIgnore) {
+				if (Thread.currentThread().getName().matches(threadToIgnore)) {
+					log.debug("Thread with name [" + Thread.currentThread().getName() +"] matches the "
```

Comment:
maybe extract Thread.currentThread().getName()? then you can use a format string (and avoid string cat when debug isn't on)